android-components/components/browser/menu2/src/test/java/mozilla/components/browser/menu2/ext/PopupWindowTest.kt

```diff
@@ -4,180 +4,388 @@
+        verify(popupWindow, never()).showAtLocation(any(), anyInt(), anyInt(), anyInt())
     }
 
     @Test
-    fun `WHEN display down popup on right to left device THEN show on the correct side of the anchor`() {
-        anchor = createMockViewWith(x = 90, y = 0, true)
-        popupWindow.displayPopup(menuContentView, anchor, Orientation.DOWN)
+    fun `WHEN recycler view has no measured height THEN show is never called`() {
+        val containerRecyclerView = createContainerRecyclerView(recyclerViewHeight = 0)
+        doReturn(containerRecyclerView).`when`(containerView)
+            .findViewById<RecyclerView>(R.id.mozac_browser_menu_recyclerView)
+
+        val containerHeight = containerRecyclerView.measuredHeight.plus(CONTAINER_PADDING)
+        `when`(containerView.measuredHeight).thenReturn(containerHeight)
 
-        assertEquals(R.style.Mozac_Browser_Menu2_Animation_OverflowMenuRightTop, popupWindow.animationStyle)
-        verify(popupWindow).showAsDropDown(anchor, 0, 0)
+        anchor = mock(View::class.java)
+
+        popupWindow.displayPopup(containerView, anchor, Orientation.UP)
+
+        verify(popupWindow, never()).showAsDropDown(any())
+        verify(popupWindow, never()).showAsDropDown(any(), anyInt(), anyInt())
+        verify(popupWindow, never()).showAtLocation(any(), anyInt(), anyInt(), anyInt())
     }
 
     @Test
-    fun `WHEN displaying prefer up popup from top right THEN show popup down and to the left`() {
-        anchor = createMockViewWith(x = 90, y = 0, false)
-        popupWindow.displayPopup(menuContentView, anchor, Orientation.UP)
+    fun `WHEN recycler view has no items THEN show is never called`() {
+        val containerRecyclerView =
+            createContainerRecyclerView(recyclerViewHeight = 100, itemCount = 0)
+        doReturn(containerRecyclerView).`when`(containerView)
+            .findViewById<RecyclerView>(R.id.mozac_browser_menu_recyclerView)
+
+        val containerHeight = containerRecyclerView.measuredHeight.plus(CONTAINER_PADDING)
+        `when`(containerView.measuredHeight).thenReturn(containerHeight)
 
-        assertEquals(R.style.Mozac_Browser_Menu2_Animation_OverflowMenuRightTop, popupWindow.animationStyle)
-        verify(popupWindow).showAsDropDown(anchor, 0, 0)
+        anchor = mock(View::class.java)
+
+        popupWindow.displayPopup(containerView, anchor, Orientation.UP)
+
+        verify(popupWindow, never()).showAsDropDown(any())
+        verify(popupWindow, never()).showAsDropDown(any(), anyInt(), anyInt())
+        verify(popupWindow, never()).showAtLocation(any(), anyInt(), anyInt(), anyInt())
     }
 
     @Test
-    fun `WHEN displaying prefer up popup from bottom right THEN show popup up and to the left`() {
-        anchor = createMockViewWith(x = 90, y = 190, false)
-        popupWindow.displayPopup(menuContentView, anchor, Orientation.UP)
+    fun `WHEN orientation up THEN showAsDropDown is called with expected config`() {
```

Comment:
Maybe we could change the ambiguous `called with expected config` THEN condition to something more specific? Like 
`WHEN orientation up & does fit available height THEN the popup keeps its height & is positioned from the bottom of the anchor with leftBottomAnimation`

It is a mouthful, but it describes the business requirement we have. It might come in handy if we change the behavior: tests will start to fail, and the name of the test would tell us exactly what is the use case we should fix.